Enable reading from zipped files

Hyralc opened the pull request.

Hyralc committed:
typo

Hyralc committed:
Read inside zipped output

pl-buiquang commented on antarest/study/storage/rawstudy/model/filesystem/ini_file_node.py:
```diff
@@ -60,10 +58,17 @@ def _get(
+            try:
+                json = self.reader.read(self.path)
+            except Exception as e:
+                raise IniReaderError(self.__class__.__name__, str(e))
```
the try catch should surround the self.reader.read in both cases (especially needed to clean up the tmp_dir if there is an exception)

Hyralc committed:
fix windows bug

Hyralc committed:
enable parsing inside zipped study

Hyralc committed:
fix for windows

Hyralc committed:
fix for windows

pl-buiquang commented on antarest/study/storage/rawstudy/model/filesystem/config/files.py:
```diff
@@ -65,8 +74,40 @@ def build(
+    ) -> Any:
+        tmp_dir = None
+        if root.suffix == ".zip":
+            from antarest.study.storage.utils import extract_file_to_tmp_dir
```
is this because import loop ? better fix the circular import (should be easy since this is just an utils function, you can put it in common/utils)

pl-buiquang commented on antarest/study/storage/rawstudy/model/filesystem/config/files.py:
```diff
@@ -184,7 +243,7 @@ def parse_simulation(path: Path) -> Optional["Simulation"]:
                 by_year=by_year,
                 synthesis=synthesis,
-                error=not (path / "checkIntegrity.txt").exists(),
+                error=error,  # TODO: does not work when zipped
```
remove todo (it works now)

pl-buiquang commented on antarest/study/service.py:
```diff
@@ -703,7 +701,7 @@ def sync_studies_on_disk(
                             id=str(uuid4()),
                             name=folder.path.name,
-                            path=str(folder.path),
+                            path=str(folder.path.parent / folder.path.stem),
```
why change this ? here all synced studies are unzipped studies

pl-buiquang commented on antarest/study/storage/rawstudy/watcher.py:
```diff
@@ -129,7 +130,17 @@ def _rec_scan(
+            elif path.suffix == ".zip":
+                zf = ZipFile(path, "r")
+                if str(Path(path.stem) / "study.antares") in zf.namelist():
+                    study_antares_exist = True
```
don't try to sync archived studies (only managed archived studies will be handled)

pl-buiquang commented on antarest/study/storage/utils.py:
```diff
@@ -345,3 +345,14 @@ def get_start_date(
     )
+
+
+def extract_file_to_tmp_dir(
```
move this is in common/utils

pl-buiquang commented on antarest/study/storage/rawstudy/raw_study_service.py:
```diff
@@ -128,7 +129,12 @@ def exists(self, metadata: RawStudy) -> bool:
         """
-        return (self.get_study_path(metadata) / "study.antares").is_file()
+        path = self.get_study_path(metadata)
+        if path.suffix == ".zip":
```
here the check should be :  
```if metadata.archived:```

Hyralc committed:
read inside zip only in managed studies

need front fix: inside the debug tab of the archived study, no request is sent to the api, which does not enable to check if the results are correct

Hyralc committed:
fix bug where outputs are extracted even when asked to stay archived

pl-buiquang committed:
Allow front to explore archived studies

Signed-off-by: Paul Bui-Quang <paul.buiquang@rte-france.com>

Hyralc committed:
done

Hyralc committed:
Merge remote-tracking branch 'origin/dev' into issue-926-Enable_reading_from_zipped_files

Hyralc committed:
fix, still need to fix fall back output import

pl-buiquang commented on antarest/study/storage/abstract_storage_service.py:
```diff
@@ -317,7 +311,7 @@ def export_study(
             stopwatch = StopWatch()
             filename = shutil.make_archive(
-                base_name=os.path.splitext(target)[0],
+                base_name=os.path.splitext(target)[0],  # TODO:MAYBE THIS
```
what do you mean by "maybe this" ?

pl-buiquang commented on antarest/study/storage/rawstudy/model/filesystem/config/files.py:
```diff
@@ -65,8 +75,38 @@ def build(
+            output_data = IniReader().read(output_data_path)
+
+        if tmp_dir:
+            tmp_dir.cleanup()
```
enclose this in a try..finally to cleanup the tmp_dir in all cases

pl-buiquang commented on antarest/study/storage/rawstudy/model/filesystem/config/files.py:
```diff
@@ -171,6 +221,11 @@ def parse_simulation(path: Path) -> Optional["Simulation"]:
         try:
+            if path.suffix == ".zip":
+                zf = ZipFile(path, "r")
+                error = str("checkIntegrity.txt") not in zf.namelist()
```
worried about the speed of zf.namelist() for big output on a network disk.
can you test it ?

pl-buiquang commented on antarest/study/storage/rawstudy/model/filesystem/config/model.py:
```diff
@@ -138,8 +139,16 @@ def __init__(
+    def next_file(
+        self, name: str, is_output: bool = False
+    ) -> "FileStudyTreeConfig":
+        if is_output and self.outputs[name].archived:
```
check if name is in outputs (had an error on some dirty study)

pl-buiquang commented on antarest/study/storage/rawstudy/watcher.py:
```diff
@@ -129,16 +129,18 @@ def _rec_scan(
                         try:
                             if (
-                                child.is_dir()
+                                (child.is_dir() or child.suffix == ".zip")
```
don't scan zipped studies

pl-buiquang commented on antarest/study/storage/rawstudy/raw_study_service.py:
```diff
@@ -349,10 +357,11 @@ def set_reference_output(
-        self.export_study(study, archive_path)
+        new_study_path = self.export_study(study, archive_path)
         shutil.rmtree(study.path)
+        return new_study_path
```
remove study from cache (if not the filestudytreeconfig stills thinks the study is unarchived and we fail to explore the study)

pl-buiquang commented on antarest/study/storage/rawstudy/raw_study_service.py:
```diff
@@ -366,6 +375,11 @@ def get_study_path(self, metadata: Study) -> Path:
+            return self.get_archive_path(metadata)
+        return Path(metadata.path)
+
+    def get_study_path_to_unarchive(self, metadata: Study) -> Path:
```
useless method

pl-buiquang reviewed (changes requested):
- Missing zip support for version (l112), binding_constraints (l142), link (l347), thermals (l317), renewables (l331) parsing in antarest/study/storage/rawstudy/model/filesystem/config/files.py 
- Remove study from cache at the end of the unarchive task

Hyralc committed:
apply PR comments

Hyralc committed:
fixes
